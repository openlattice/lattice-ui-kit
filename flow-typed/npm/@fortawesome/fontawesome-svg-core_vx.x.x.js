// flow-typed signature: 39887e5e6c6c32861704232d9719cd14
// flow-typed version: <<STUB>>/@fortawesome/fontawesome-svg-core_v1.2.4/flow_v0.80.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@fortawesome/fontawesome-svg-core'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@fortawesome/fontawesome-svg-core' {
  declare module.exports: any;
}
