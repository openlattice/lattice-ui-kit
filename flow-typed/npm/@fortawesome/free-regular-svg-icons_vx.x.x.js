// flow-typed signature: e06b1e176fa03c6cedb9f03bb6cc7872
// flow-typed version: <<STUB>>/@fortawesome/free-regular-svg-icons_v5.13.0/flow_v0.118.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@fortawesome/free-regular-svg-icons'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@fortawesome/free-regular-svg-icons' {
  declare module.exports: any;
}
