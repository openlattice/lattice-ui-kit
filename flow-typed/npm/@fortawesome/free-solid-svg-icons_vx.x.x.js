// flow-typed signature: 64f61153e222e791ad6eca6c60c55baa
// flow-typed version: <<STUB>>/@fortawesome/free-solid-svg-icons_v5.13.0/flow_v0.118.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@fortawesome/free-solid-svg-icons'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@fortawesome/free-solid-svg-icons' {
  declare module.exports: any;
}
