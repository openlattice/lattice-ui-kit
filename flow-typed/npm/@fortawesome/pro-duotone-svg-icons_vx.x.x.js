// flow-typed signature: dc91210f6aed06e504e8c5d05af6775b
// flow-typed version: <<STUB>>/@fortawesome/pro-duotone-svg-icons_v5.10.2/flow_v0.106.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@fortawesome/pro-duotone-svg-icons'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@fortawesome/pro-duotone-svg-icons' {
  declare module.exports: any;
}
