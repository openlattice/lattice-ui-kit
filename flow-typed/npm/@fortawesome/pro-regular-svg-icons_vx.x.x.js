// flow-typed signature: c41877c2f635a0484c823342d488e66e
// flow-typed version: <<STUB>>/@fortawesome/pro-regular-svg-icons_v5.3.1/flow_v0.80.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@fortawesome/pro-regular-svg-icons'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@fortawesome/pro-regular-svg-icons' {
  declare module.exports: any;
}
