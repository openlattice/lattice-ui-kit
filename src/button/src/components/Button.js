/*
 * @flow
 */

import React, { Component } from 'react';
import type { Node } from 'react';

import PropTypes from 'prop-types';

import DefaultButton from './styled/DefaultButton';
import PrimaryButton from './styled/PrimaryButton';
import SecondaryButton from './styled/SecondaryButton';

type ButtonMode =
  | 'default'
  | 'primary'
  | 'secondary';

type Props = {
  children :Node;
  className ? :string;
  disabled ? :boolean;
  mode ? :ButtonMode;
  onClick :(e :SyntheticEvent<HTMLButtonElement>) => void;
  type ? :string;
};

type State = {};

/*
 * Inspiration:
 * https://atlaskit.atlassian.com/packages/core/button
 * https://evergreen.surge.sh/components/buttons
 */
export default class Button extends Component<Props, State> {

  static propTypes = {
    children: PropTypes.node.isRequired,
    className: PropTypes.string,
    disabled: PropTypes.bool,
    mode: PropTypes.string,
    onClick: PropTypes.func.isRequired,
    type: PropTypes.string,
  }

  static defaultProps = {
    className: '',
    disabled: false,
    mode: 'default',
    type: 'button',
  }

  render() {

    const {
      children,
      className,
      disabled,
      mode,
      onClick,
      type,
    } = this.props;

    switch (mode) {
      case 'primary':
        return (
          <PrimaryButton className={className} disabled={disabled} onClick={onClick} type={type}>
            { children }
          </PrimaryButton>
        );
      case 'secondary':
        return (
          <SecondaryButton className={className} disabled={disabled} onClick={onClick} type={type}>
            { children }
          </SecondaryButton>
        );
      default:
        return (
          <DefaultButton className={className} disabled={disabled} onClick={onClick} type={type}>
            { children }
          </DefaultButton>
        );
    }
  }
}
