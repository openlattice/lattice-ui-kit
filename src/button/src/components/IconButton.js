/*
 * @flow
 */

import React, { useMemo } from 'react';

import clsx from 'clsx';
import styled from 'styled-components';
import { IconButton as MuiIconButton } from '@material-ui/core';

import useButtonStyles, { isCustomColor } from './useButtonStyles';
import { styleName } from './createColorStyles';
import type { ButtonProps } from './types';

import Spinner from '../../../spinner';

const IconSquareWrapper = styled.div`
  align-items: center;
  display: flex;
  justify-content: center;
  width: auto;

  &::before {
    content: "";
    display: block;
    padding-bottom: 100%; /* padding size is relative to the width */
  }
`;

const SpinnerWrapper = styled.div`
  align-items: center;
  background: white;
  border-radius: 50%;
  display: flex;
  height: 100%;
  justify-content: center;
  position: absolute;
  width: 100%;
`;

const IconButton = ({
  children,
  className,
  color = 'default',
  disabled = false,
  isLoading = false,
  ...rest
} :ButtonProps) => {

  const classes = useButtonStyles();
  const customColor = useMemo(() => isCustomColor(color), [color]);

  // https://medium.com/flow-type/spreads-common-errors-fixes-9701012e9d58
  // [...] you can annotate the object with optional properties and then explicitly set the property afterwards
  const classesObject = {
    [classes[styleName('text', color)]]: customColor,
  };
  if (className) {
    classesObject[className] = className;
  }

  const props = {
    ...rest,
    className: clsx(classesObject),
    color: customColor ? 'default' : color,
    disabled: disabled || isLoading,
  };

  /* eslint-disable react/jsx-props-no-spreading */
  return (
    <MuiIconButton {...props}>
      <IconSquareWrapper>
        {
          isLoading && (
            <SpinnerWrapper>
              <Spinner />
            </SpinnerWrapper>
          )
        }
        {children}
      </IconSquareWrapper>
    </MuiIconButton>
  );
  /* eslint-enable */
};

export default IconButton;
