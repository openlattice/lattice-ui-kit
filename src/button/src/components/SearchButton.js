/*
 * @flow
 */

import React from 'react';

import { faSearch } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import IconButton from './IconButton';
import type { Props } from './IconButton';

const icon = (
  <FontAwesomeIcon icon={faSearch} />
);

/* eslint-disable react/jsx-props-no-spreading */
const SearchButton = (props :Props) => (
  <IconButton icon={icon} {...props} />
);
/* eslint-enable */

export default SearchButton;
