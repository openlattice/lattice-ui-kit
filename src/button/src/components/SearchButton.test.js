import React from 'react';
import toJson from 'enzyme-to-json';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { mount } from 'enzyme';

import Button from './Button';
import IconButton from './IconButton';
import SearchButton from './SearchButton';

const BUTTON_TXT = 'Search';

describe('SearchButton', () => {

  describe('icon with text', () => {

    const searchButton = mount(
      <SearchButton>{BUTTON_TXT}</SearchButton>
    );

    test('should match snapshot', () => {
      expect(toJson(searchButton)).toMatchSnapshot();
    });

    test('should render a button element', () => {
      expect(searchButton.find('button')).toHaveLength(1);
    });

    test('should render FontAwesomeIcon component', () => {
      expect(searchButton.find(FontAwesomeIcon)).toHaveLength(1);
    });

    test('should render IconButton component', () => {
      expect(searchButton.find(IconButton)).toHaveLength(1);
    });

    test('should render Button component', () => {
      expect(searchButton.find(Button)).toHaveLength(1);
    });

    test('should render the correct text', () => {
      expect(searchButton.text()).toEqual(BUTTON_TXT);
    });

  });

  describe('icon without text', () => {

    const searchButton = mount(
      <SearchButton />
    );

    test('should match snapshot', () => {
      expect(toJson(searchButton)).toMatchSnapshot();
    });

    test('should render a button element', () => {
      expect(searchButton.find('button')).toHaveLength(1);
    });

    test('should render FontAwesomeIcon component', () => {
      expect(searchButton.find(FontAwesomeIcon)).toHaveLength(1);
    });

    test('should render IconButton component', () => {
      expect(searchButton.find(IconButton)).toHaveLength(1);
    });

    test('should render Button component', () => {
      expect(searchButton.find(Button)).toHaveLength(1);
    });

    test('should not render any text', () => {
      expect(searchButton.text()).toEqual('');
    });

  });

});
