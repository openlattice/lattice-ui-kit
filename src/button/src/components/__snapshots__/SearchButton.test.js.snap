// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchButton icon with text should match snapshot 1`] = `
.c2 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  opacity: 1;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c1 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  max-width: 100%;
  position: relative;
  overflow: hidden;
  min-width: 14px;
}

.c0 {
  border-radius: 3px;
  border-style: solid;
  border-width: 1px;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 14px;
  line-height: 18px;
  outline: none;
  padding: 10px 20px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: background-color 100ms ease-out,border-color 100ms ease-out,box-shadow 100ms ease-out;
  transition: background-color 100ms ease-out,border-color 100ms ease-out,box-shadow 100ms ease-out;
  white-space: nowrap;
  background-color: #f2f2f4;
  border-color: #f2f2f4;
  color: #56565a;
}

.c0:hover {
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #dfdfe3;
  border-color: #dfdfe3;
}

.c0:active {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #d1d1d6;
  border-color: #d1d1d6;
  color: #56565a;
}

.c0:disabled {
  cursor: not-allowed;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #f9f9fc;
  border-color: #f9f9fc;
  color: #8b8b90;
}

.c0:focus {
  box-shadow: #d1d1d6 0 0 0 2px;
}

.c3 {
  margin: 0 8px 0 0;
}

<SearchButton>
  <IconButton
    icon={
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f002",
              "M508.5 468.9L387.1 347.5c-2.3-2.3-5.3-3.5-8.5-3.5h-13.2c31.5-36.5 50.6-84 50.6-136C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c52 0 99.5-19.1 136-50.6v13.2c0 3.2 1.3 6.2 3.5 8.5l121.4 121.4c4.7 4.7 12.3 4.7 17 0l22.6-22.6c4.7-4.7 4.7-12.3 0-17zM208 368c-88.4 0-160-71.6-160-160S119.6 48 208 48s160 71.6 160 160-71.6 160-160 160z",
            ],
            "iconName": "search",
            "prefix": "far",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    }
  >
    <Button
      disabled={false}
      fontColor=""
      isLoading={false}
      mode="default"
      type="button"
    >
      <StyledButton
        disabled={false}
        fontColor=""
        mode="default"
        type="button"
      >
        <StyledComponent
          disabled={false}
          fontColor=""
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "StyledButton-sc-1etazn9-0",
                "isStatic": false,
                "lastClassName": "c0",
                "rules": Array [
                  "border-radius:3px;border-style:solid;border-width:1px;box-sizing:border-box;cursor:pointer;font-size:14px;line-height:18px;outline:none;padding:10px 20px;text-align:center;text-decoration:none;transition:background-color ",
                  "100ms",
                  " ease-out,border-color ",
                  "100ms",
                  " ease-out,box-shadow ",
                  "100ms",
                  " ease-out;width:",
                  [Function],
                  ";white-space:nowrap;",
                  [Function],
                  ";",
                  [Function],
                  ";:hover{cursor:pointer;text-decoration:none;",
                  [Function],
                  "};:active{text-decoration:none;",
                  [Function],
                  "};:disabled{cursor:not-allowed;text-decoration:none;",
                  [Function],
                  "};:focus{",
                  [Function],
                  "}",
                ],
              },
              "displayName": "StyledButton",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "StyledButton-sc-1etazn9-0",
              "target": "button",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
          mode="default"
          type="button"
        >
          <button
            className="c0"
            disabled={false}
            mode="default"
            type="button"
          >
            <ContentWrapper>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "ContentWrapper-sc-36hfub-0",
                      "isStatic": false,
                      "lastClassName": "c1",
                      "rules": Array [
                        "align-self:center;display:inline-flex;flex-wrap:nowrap;max-width:100%;position:relative;overflow:hidden;min-width:14px;",
                      ],
                    },
                    "displayName": "ContentWrapper",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "ContentWrapper-sc-36hfub-0",
                    "target": "span",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <span
                  className="c1"
                >
                  <Content
                    isLoading={false}
                  >
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "Content-sc-18psqsj-0",
                            "isStatic": false,
                            "lastClassName": "c2",
                            "rules": Array [
                              "flex:1 1 auto;opacity:",
                              [Function],
                              ";overflow:hidden;text-overflow:ellipsis;white-space:nowrap;",
                            ],
                          },
                          "displayName": "Content",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "Content-sc-18psqsj-0",
                          "target": "span",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      isLoading={false}
                    >
                      <span
                        className="c2"
                      >
                        <IconButton__IconMarginRight>
                          <StyledComponent
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "IconButton__IconMarginRight-g9q5gl-0",
                                  "isStatic": false,
                                  "lastClassName": "c3",
                                  "rules": Array [
                                    "margin:0 8px 0 0;",
                                  ],
                                },
                                "displayName": "IconButton__IconMarginRight",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "IconButton__IconMarginRight-g9q5gl-0",
                                "target": "span",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                          >
                            <span
                              className="c3"
                            >
                              <FontAwesomeIcon
                                border={false}
                                className=""
                                fixedWidth={false}
                                flip={null}
                                icon={
                                  Object {
                                    "icon": Array [
                                      512,
                                      512,
                                      Array [],
                                      "f002",
                                      "M508.5 468.9L387.1 347.5c-2.3-2.3-5.3-3.5-8.5-3.5h-13.2c31.5-36.5 50.6-84 50.6-136C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c52 0 99.5-19.1 136-50.6v13.2c0 3.2 1.3 6.2 3.5 8.5l121.4 121.4c4.7 4.7 12.3 4.7 17 0l22.6-22.6c4.7-4.7 4.7-12.3 0-17zM208 368c-88.4 0-160-71.6-160-160S119.6 48 208 48s160 71.6 160 160-71.6 160-160 160z",
                                    ],
                                    "iconName": "search",
                                    "prefix": "far",
                                  }
                                }
                                inverse={false}
                                listItem={false}
                                mask={null}
                                pull={null}
                                pulse={false}
                                rotation={null}
                                size={null}
                                spin={false}
                                swapOpacity={false}
                                symbol={false}
                                title=""
                                transform={null}
                              >
                                <svg
                                  aria-hidden="true"
                                  className="svg-inline--fa fa-search fa-w-16 "
                                  data-icon="search"
                                  data-prefix="far"
                                  focusable="false"
                                  role="img"
                                  style={Object {}}
                                  viewBox="0 0 512 512"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M508.5 468.9L387.1 347.5c-2.3-2.3-5.3-3.5-8.5-3.5h-13.2c31.5-36.5 50.6-84 50.6-136C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c52 0 99.5-19.1 136-50.6v13.2c0 3.2 1.3 6.2 3.5 8.5l121.4 121.4c4.7 4.7 12.3 4.7 17 0l22.6-22.6c4.7-4.7 4.7-12.3 0-17zM208 368c-88.4 0-160-71.6-160-160S119.6 48 208 48s160 71.6 160 160-71.6 160-160 160z"
                                    fill="currentColor"
                                    style={Object {}}
                                  />
                                </svg>
                              </FontAwesomeIcon>
                            </span>
                          </StyledComponent>
                        </IconButton__IconMarginRight>
                        Search
                      </span>
                    </StyledComponent>
                  </Content>
                </span>
              </StyledComponent>
            </ContentWrapper>
          </button>
        </StyledComponent>
      </StyledButton>
    </Button>
  </IconButton>
</SearchButton>
`;

exports[`SearchButton icon without text should match snapshot 1`] = `
.c2 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  opacity: 1;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c1 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  max-width: 100%;
  position: relative;
  overflow: hidden;
  min-width: 14px;
}

.c0 {
  border-radius: 3px;
  border-style: solid;
  border-width: 1px;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 14px;
  line-height: 18px;
  outline: none;
  padding: 10px 20px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: background-color 100ms ease-out,border-color 100ms ease-out,box-shadow 100ms ease-out;
  transition: background-color 100ms ease-out,border-color 100ms ease-out,box-shadow 100ms ease-out;
  white-space: nowrap;
  background-color: #f2f2f4;
  border-color: #f2f2f4;
  color: #56565a;
}

.c0:hover {
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #dfdfe3;
  border-color: #dfdfe3;
}

.c0:active {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #d1d1d6;
  border-color: #d1d1d6;
  color: #56565a;
}

.c0:disabled {
  cursor: not-allowed;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #f9f9fc;
  border-color: #f9f9fc;
  color: #8b8b90;
}

.c0:focus {
  box-shadow: #d1d1d6 0 0 0 2px;
}

<SearchButton>
  <IconButton
    icon={
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f002",
              "M508.5 468.9L387.1 347.5c-2.3-2.3-5.3-3.5-8.5-3.5h-13.2c31.5-36.5 50.6-84 50.6-136C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c52 0 99.5-19.1 136-50.6v13.2c0 3.2 1.3 6.2 3.5 8.5l121.4 121.4c4.7 4.7 12.3 4.7 17 0l22.6-22.6c4.7-4.7 4.7-12.3 0-17zM208 368c-88.4 0-160-71.6-160-160S119.6 48 208 48s160 71.6 160 160-71.6 160-160 160z",
            ],
            "iconName": "search",
            "prefix": "far",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    }
  >
    <Button
      disabled={false}
      fontColor=""
      isLoading={false}
      mode="default"
      type="button"
    >
      <StyledButton
        disabled={false}
        fontColor=""
        mode="default"
        type="button"
      >
        <StyledComponent
          disabled={false}
          fontColor=""
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "StyledButton-sc-1etazn9-0",
                "isStatic": false,
                "lastClassName": "c0",
                "rules": Array [
                  "border-radius:3px;border-style:solid;border-width:1px;box-sizing:border-box;cursor:pointer;font-size:14px;line-height:18px;outline:none;padding:10px 20px;text-align:center;text-decoration:none;transition:background-color ",
                  "100ms",
                  " ease-out,border-color ",
                  "100ms",
                  " ease-out,box-shadow ",
                  "100ms",
                  " ease-out;width:",
                  [Function],
                  ";white-space:nowrap;",
                  [Function],
                  ";",
                  [Function],
                  ";:hover{cursor:pointer;text-decoration:none;",
                  [Function],
                  "};:active{text-decoration:none;",
                  [Function],
                  "};:disabled{cursor:not-allowed;text-decoration:none;",
                  [Function],
                  "};:focus{",
                  [Function],
                  "}",
                ],
              },
              "displayName": "StyledButton",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "StyledButton-sc-1etazn9-0",
              "target": "button",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
          mode="default"
          type="button"
        >
          <button
            className="c0"
            disabled={false}
            mode="default"
            type="button"
          >
            <ContentWrapper>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "ContentWrapper-sc-36hfub-0",
                      "isStatic": false,
                      "lastClassName": "c1",
                      "rules": Array [
                        "align-self:center;display:inline-flex;flex-wrap:nowrap;max-width:100%;position:relative;overflow:hidden;min-width:14px;",
                      ],
                    },
                    "displayName": "ContentWrapper",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "ContentWrapper-sc-36hfub-0",
                    "target": "span",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <span
                  className="c1"
                >
                  <Content
                    isLoading={false}
                  >
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "Content-sc-18psqsj-0",
                            "isStatic": false,
                            "lastClassName": "c2",
                            "rules": Array [
                              "flex:1 1 auto;opacity:",
                              [Function],
                              ";overflow:hidden;text-overflow:ellipsis;white-space:nowrap;",
                            ],
                          },
                          "displayName": "Content",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "Content-sc-18psqsj-0",
                          "target": "span",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      isLoading={false}
                    >
                      <span
                        className="c2"
                      >
                        <FontAwesomeIcon
                          border={false}
                          className=""
                          fixedWidth={false}
                          flip={null}
                          icon={
                            Object {
                              "icon": Array [
                                512,
                                512,
                                Array [],
                                "f002",
                                "M508.5 468.9L387.1 347.5c-2.3-2.3-5.3-3.5-8.5-3.5h-13.2c31.5-36.5 50.6-84 50.6-136C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c52 0 99.5-19.1 136-50.6v13.2c0 3.2 1.3 6.2 3.5 8.5l121.4 121.4c4.7 4.7 12.3 4.7 17 0l22.6-22.6c4.7-4.7 4.7-12.3 0-17zM208 368c-88.4 0-160-71.6-160-160S119.6 48 208 48s160 71.6 160 160-71.6 160-160 160z",
                              ],
                              "iconName": "search",
                              "prefix": "far",
                            }
                          }
                          inverse={false}
                          listItem={false}
                          mask={null}
                          pull={null}
                          pulse={false}
                          rotation={null}
                          size={null}
                          spin={false}
                          swapOpacity={false}
                          symbol={false}
                          title=""
                          transform={null}
                        >
                          <svg
                            aria-hidden="true"
                            className="svg-inline--fa fa-search fa-w-16 "
                            data-icon="search"
                            data-prefix="far"
                            focusable="false"
                            role="img"
                            style={Object {}}
                            viewBox="0 0 512 512"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M508.5 468.9L387.1 347.5c-2.3-2.3-5.3-3.5-8.5-3.5h-13.2c31.5-36.5 50.6-84 50.6-136C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c52 0 99.5-19.1 136-50.6v13.2c0 3.2 1.3 6.2 3.5 8.5l121.4 121.4c4.7 4.7 12.3 4.7 17 0l22.6-22.6c4.7-4.7 4.7-12.3 0-17zM208 368c-88.4 0-160-71.6-160-160S119.6 48 208 48s160 71.6 160 160-71.6 160-160 160z"
                              fill="currentColor"
                              style={Object {}}
                            />
                          </svg>
                        </FontAwesomeIcon>
                      </span>
                    </StyledComponent>
                  </Content>
                </span>
              </StyledComponent>
            </ContentWrapper>
          </button>
        </StyledComponent>
      </StyledButton>
    </Button>
  </IconButton>
</SearchButton>
`;
